listenbrainz_spark/recommendations/recording/user_similarity.py

```diff
@@ -0,0 +1,98 @@
+import sys
+import math
+
+from flask import current_app
+from pyspark.mllib.linalg.distributed import CoordinateMatrix, MatrixEntry
+from pyspark.ml.stat import Correlation
+from pyspark.sql.functions import struct, collect_list
+
+import listenbrainz_spark
+from listenbrainz_spark import SparkSessionNotInitializedException, utils, path
+from listenbrainz_spark.exceptions import PathNotFoundException, FileNotFetchedException
+
+
+def create_messages(similar_users_df):
+    itr = similar_users_df.toLocalIterator()
+    message = {}
+    for row in itr:
+        message[row.user_name] = {user.other_user_name: user.similarity for user in row.similar_users}
+    yield {
+        'type': 'similar_users',
+        'data': message
+    }
+
+
+def threshold_similar_users(matrix, threshold):
+    rows, cols = matrix.shape
+    similar_users = list()
+
+    max_similarity = None
```

Comment:
style thing, but maybe we should consider extracting this logic for getting max and min similarity into a separate function. Or at least a comment here explaining that's what we're doing.

Reply:
I don't really feel that this is needed -- this code is pretty straightforward.

Reply:
I added some comments which should help here.